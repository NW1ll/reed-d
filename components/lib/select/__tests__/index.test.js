import rdSelect from "../src/main.vue";
import { mount } from "@vue/test-utils";
import mountTest from "../../../../tests/shared/mountTest";
import { ref } from "vue";
const value = ref(["song0", "song3", "song4"]);
const options = [
  {
    label: "Everybody's Got Something to Hide Except Me and My Monkey",
    value: "song0",
    disabled: true,
  },
  {
    label: "Drive My Car",
    value: "song1",
  },
  {
    label: "Norwegian Wood",
    value: "song2",
  },
  {
    label: "You Won't See",
    value: "song3",
    disabled: true,
  },
  {
    label: "Nowhere Man",
    value: "song4",
  },
  {
    label: "Think For Yourself",
    value: "song5",
  },
  {
    label: "The Word",
    value: "song6",
  },
  {
    label: "Michelle",
    value: "song7",
    disabled: true,
  },
  {
    label: "What goes on",
    value: "song8",
  },
  {
    label: "Girl",
    value: "song9",
  },
  {
    label: "I'm looking through you",
    value: "song10",
  },
  {
    label: "In My Life",
    value: "song11",
  },
  {
    label: "Wait",
    value: "song12",
  },
];

describe("rdSelect", () => {
  mountTest(rdSelect);
  it("测试渲染是否正确", () => {
    const wrapper = mount({
      render() {
        return <rdSelect>Follow</rdSelect>;
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("test props", () => {
    const wrapper = mount(rdSelect, {
      props: {
        value: value,
        options: options,
        remote: false,
        multiple: true,
        clearable: true,
      },
    });
    expect(wrapper.props()).toEqual({
      value: value,
      options: options,
      remote: false,
      multiple: true,
      clearable: true,
    });
  });
});
