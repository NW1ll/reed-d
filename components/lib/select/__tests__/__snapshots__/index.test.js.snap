// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rdSelect 测试渲染是否正确 1`] = `
<div class="rd-select is-visible" tabindex="-1">
  <div class="rd-base-select" tabindex="-1">
    <div class="rd-select-label" style="display: none;"></div>
    <div class="rd-select-placeholder">请选择</div>
    <jw-icon size="18" class="rd-select-suffix" color="rgba(194, 194, 194, 1)"><svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" viewBox="0 0 512 512" enable-background="new 0 0 512 512" xml:space="preserve">
        <path d="M256,294.1L383,167c9.4-9.4,24.6-9.4,33.9,0s9.3,24.6,0,34L273,345c-9.1,9.1-23.7,9.3-33.1,0.7L95,201.1
	c-4.7-4.7-7-10.9-7-17c0-6.1,2.3-12.3,7-17c9.4-9.4,24.6-9.4,33.9,0L256,294.1z"></path>
      </svg></jw-icon>
  </div>
  <!--v-if-->
  <div class="rd-select-dropdown">
    <div class="no-options">无选项</div>
  </div>
</div>
`;
